### ISSUE TITLE
Automating `EnforcedHashRocketStyle` and `EnforcedColonStyle`.

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15465]
### Before submitting

- [x] I have searched the existing issues and this hasn't been requested yet.

### Cop name

Layout/HashAlignment

### Is your feature request related to a problem?

In the experience of the undersigned, the default (`key`) is correct for almost all cases. 

Style `table` only seems to make sense when:

1. There is symmetry in values
2. It is important to highlight this symmetry

Style `separator` just feels fringe.

### Describe the solution you'd like

Derived indentation automatically from each context (like aligned comments), rather than being a global options.

Requesting dev comments on to what extent the relevant symmetry is detectable to a perfunctory level. 

### Code examples

```ruby
# Default (bit messy):

{
  'Emos listening to The Cure' => 'index_gour7IuV'
  'Hipsters who think they are unique' => 'index_ooW6mohd',
  'Hot chicks' => 'index_lahZ0sho',
  'Damn Hippies!!!!' => 'index_fai9aiNa',
}

# Indenting points out the common prefix and length:

{
  'Emos listening to The Cure'         => 'index_gour7IuV'
  'Hipsters who think they are unique' => 'index_ooW6mohd',
  'Hot chicks'                         => 'index_lahZ0sho',
  'Damn Hippies!!!!'                   => 'index_fai9aiNa',
}

# Notice how, without the prefix, indenting becomes a *bit* less called for:

{
  'Emos listening to The Cure' => 'gour7IuV'
  'Hipsters who think they are unique' => 'ooW6mohd',
  'Hot chicks' => 'lahZ0sho',
  'Damn Hippies!!!!' => 'fai9aiNa',
}

# For keys with similar lengths, sort by length can be a satisfying compromise.

{
  'Emos' => 'index_gour7IuV'
  'Chicks' => 'index_lahZ0sho',
  'Hippies' => 'index_fai9aiNa',
  'Hipsters' => 'index_ooW6mohd',
}
```

### Describe alternatives you've considered

Nothing except disabling in config or code

### Additional context

For comparison, indenting is hardly ever seen in CSS, even though you'd think so:

```css
p {
  border-radius: 16px;
  min-width: 16px;
  height: 16px;
  line-height: 16px;
  font-size: 16px;
  padding: 16px;
}
```

**Comment by forthrin** [comment]
Again, AI classifies this as excessive prettifying and suggests current default and options are good. The undersigned concurs.
